### DISCUSSION TITLE
postgres comment

### DISCUSSION THREAD

**Discussion by EurekaJiang** [dbeaver/dbeaver#18000] [Q&A]
It seems that all the descriptions are saved in pg_description from the postgresql documentation.
I  wonder the description in the postgres  is selected separately and then put it into the results in DBeaver? 
I can see the columns comments in the table ,but cannot see in the view .

![NBU0%`3~)%JQD1N}1FJUSU](https://user-images.githubusercontent.com/41953750/197442403-910a71e4-0d20-4d8c-9d5e-8e37415a638c.png)

![ZGP}X R{AVEXDS%0V831XOA](https://user-images.githubusercontent.com/41953750/197442430-66601b8c-fab1-4b5e-bfa7-7e57897be7e6.jpg)

**Comment by LonwoLonwo** [comment]
Hello @Jtworld0924 

Usually, DBeaver uses some query like this for columns metadata reading:

```sql
SELECT c.relname,a.*,pg_catalog.pg_get_expr(ad.adbin, ad.adrelid, true) as def_value,dsc.description
FROM pg_catalog.pg_attribute a
INNER JOIN pg_catalog.pg_class c ON (a.attrelid=c.oid)
LEFT OUTER JOIN pg_catalog.pg_attrdef ad ON (a.attrelid=ad.adrelid AND a.attnum = ad.adnum)
LEFT OUTER JOIN pg_catalog.pg_description dsc ON (c.oid=dsc.objoid AND a.attnum = dsc.objsubid)
WHERE NOT a.attisdropped AND c.oid=<table.id> ORDER BY a.attnum
```

So yes, for the columns of the view we are reading descriptions from the pg_description. And I can see descriptions in my PG views columns right now. So. Maybe refresh can help in your case?

**Reply by EurekaJiang** [reply]
How thoughtful of you!  I will try it later
